test/charms/test_main/src/charm.py

```diff
@@ -26,6 +26,12 @@
 
 logger = logging.getLogger()
 
+# Note for ops developers: juju-log doesn't go anywhere useful much of the time
+# during unit tests, and test_main failures that subprocess out are often
+# difficult to debug. Uncomment this line to get more informative errors when
+# running the tests.
+# logger.addHandler(logging.StreamHandler(sys.stderr))
```

Comment:
Thanks for this.
I've found out the hard way while debugging `ops[tracing]` test failures.